control_panel_api/permissions.py

```diff
@@ -28,5 +31,23 @@ class S3BucketPermissions(IsSuperuser):
     pass
 
 
-class UserPermissions(IsSuperuser):
+class IsSelf(BasePermission):
+
+    def has_permission(self, request, view):
+        if super().has_permission(request, view):
+            return True
+
+        if hasattr(request.user, 'auth0_id'):
+            return view.kwargs.get('pk') == request.user.auth0_id
```

Comment:
Don't this break the listing of users?

e.g. this is called on all endpoints (on the list one as well), when I'm making a request to `GET /users`, `view.kwargs.get('pk')` will probably be `None` (or something like that) therefore will not be equal to `request.user.auth0_id`.

In addition, as it's not a member endpoint (as it's not operating on a specific object) the `has_object_permission()` should not be called.

Reply:
This is intentional. Normal users are not allowed to list users. I don't think it's useful to `GET /users` and only see yourself listed.

The `has_object_permission` will not be called, because it is not a member endpoint.